www/settings/inner.js

```diff
@@ -1569,6 +1570,62 @@ define([
         return $div;
     };
 
+    create['code-max-width'] = function () {
+        var key = 'hardWrapMaxWidth';
+        var hardWrapMaxWidthEnabledKey = `${key}Enabled`;
+        var defaultInitialMaxWidth = 60;
+
+        var $div = $('<div>', {
+            'class': 'cp-settings-code-max-width cp-sidebarlayout-element'
+        });
+
+        $('<label>').text(Messages.settings_codeMaxWidth).appendTo($div);
+        $('<span>', {'class': 'cp-sidebarlayout-description'})
+            .text(Messages.settings_codeMaxWidthDescription).appendTo($div);
+
+        var $container = $('<span>', {
+            'class': 'cp-sidebarlayout-input-block'
+        }).appendTo($div);
+
+        var $cbox = $(UI.createCheckbox('cp-settings-code-max-width'))
+        $cbox.appendTo($container);
+
+        var $input = $('<input>', {
+            'min': defaultInitialMaxWidth,
+            type: 'number',
+        })
+        .on('change', function () {
+            var val = parseInt($input.val());
+            if (!isNaN(val) && typeof (val) !== 'number') { return; }
+            common.setAttribute(['codemirror', key], val);
+        }).appendTo($container);
+
+        $cbox.find('input').change(
+            function () {
+                // If enable hard wrap max width is enabled, then disabled=false
+                // otherwise, disabled=true 
+                $input.attr('disabled', !this.checked);
+                common.setAttribute(['codemirror', hardWrapMaxWidthEnabledKey], this.checked);
```

Comment:
Made it so that when this feature is disabled, the input to change the width for wrapping is also disabled, and vice versa.